.github/templates/linux_ci_workflow.yml.in

```diff
@@ -23,17 +23,59 @@ env:
 
 jobs:
   calculate-docker-image:
-    runs-on: ubuntu-18.04
+    runs-on: linux.2xlarge
+    env:
+      DOCKER_BUILDKIT: 1
+    timeout-minutes: 90
     outputs:
       docker_image: ${{ steps.calculate-tag.outputs.docker_image }}
     steps:
       - name: Checkout PyTorch
         uses: actions/checkout@v2
+        with:
+          # deep clone, to allow use of git merge-base
+          fetch-depth: 0
       - name: Calculate docker image tag
         id: calculate-tag
         run: |
           DOCKER_TAG=$(git rev-parse HEAD:.circleci/docker)
+          echo "::set-output name=docker_tag::${DOCKER_TAG}"
           echo "::set-output name=docker_image::${DOCKER_IMAGE_BASE}:${DOCKER_TAG}"
+      - name: Check if image should be built
+        id: check
+        env:
```

Comment:
question, do we know why we dont need the `AWS_*` environment variable here?

Reply:
yeah afaik it's because our custom runners have IAM access automatically; @seemethere knows more about this though